src/accelerate/utils/bitsandbytes.py

```diff
@@ -0,0 +1,332 @@
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+
+import torch
+import torch.nn as nn
+
+from .imports import (
+    is_4bit_bnb_available,
+    is_8bit_bnb_available,
+    is_bnb_available,
+    is_peft_available,
+    is_transformers_available,
+)
+
+
+if is_bnb_available():
+    import bitsandbytes as bnb
+
+logger = logging.getLogger(__name__)
+
+
+def get_bnb_model(model, bnb_config, is_peft_model):
+    """
+        This function will quantize the input model with the parameters specified in bnb_config.
+
+    Args:
+        model (`torch.nn.Module`): Input model
+        bnb_config (`dict`'): The bitsandbytes parameters
+
+    Returns:
+        `torch.nn.Module`: The quantized model
+    """
+    if is_transformers_available():
+        from transformers import BitsAndBytesConfig
+        from transformers.utils.bitsandbytes import get_keys_to_not_convert
```

Comment:
You should raise an error here if transformers is not available. But do we really need the config from Transformers? Can't the plugin be used instead?

Also is `get_keys_to_not_convert` general enough to work for non-transformer models?